core/trino-main/src/test/java/io/trino/cost/TestFilterProjectAggregationStatsRule.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cost;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.iterative.rule.test.PlanBuilder;
+import io.trino.sql.planner.plan.PlanNode;
+import org.testng.annotations.Test;
+
+import java.util.function.Function;
+
+import static io.trino.SystemSessionProperties.NON_ESTIMATABLE_PREDICATE_APPROXIMATION_ENABLED;
+import static io.trino.cost.FilterStatsCalculator.UNKNOWN_FILTER_COEFFICIENT;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.sql.planner.iterative.rule.test.PlanBuilder.expression;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public class TestFilterProjectAggregationStatsRule
+        extends BaseStatsCalculatorTest
+{
+    @Test
+    public void testFilterOverAggregationStats()
```

Comment:
Add test case with pruning projection in between filter and aggregation.

Add a test when rule does not fire for non-identity projection